File: Backend/src/main/java/ir/netpick/mailmine/email/service/EmailService.java
package ir.netpick.mailmine.email.service;

import ir.netpick.mailmine.email.dto.EmailRequest;

import java.util.Map;

public interface EmailService {

    /**
     * Send a simple text email
     */
    void sendSimpleMail(EmailRequest request);

    /**
     * Send an email with attachment
     */
    void sendMailWithAttachment(EmailRequest request);

    /**
     * Send an HTML email using a Thymeleaf template
     */
    void sendTemplatedEmail(String recipient, String subject, String templateName, Map<String, Object> variables);

    /**
     * Send verification email to a user (for auth)
     */
    void sendVerificationEmail(String email, String code, int expirationMinutes);

    /**
     * Send mass emails to multiple recipients
     */
    void sendMassEmail(EmailRequest request);
}

File: Frontend/src/services/emailService.ts
// emailService.ts
import api from './api';

export interface EmailRequest {
  recipient?: string;
  subject: string;
  body: string;
  attachment?: string;
  recipients?: string[];
  templateName?: string;
}

class EmailService {
  private basePath = '/api/v1/email';

  /**
   * Send a simple email
   */
  public async sendEmail(request: EmailRequest): Promise<{ message: string }> {
    return await api.post<{ message: string }>(`${this.basePath}/send`, request);
  }

  /**
   * Send an email with attachment
   */
  public async sendEmailWithAttachment(request: EmailRequest): Promise<{ message: string }> {
    return await api.post<{ message: string }>(`${this.basePath}/send-with-attachment`, request);
  }

  /**
   * Send mass emails to multiple recipients
   */
  public async sendMassEmail(request: EmailRequest): Promise<{ message: string; recipientCount: string }> {
    return await api.post<{ message: string; recipientCount: string }>(`${this.basePath}/send-mass`, request);
  }
}

const emailService = new EmailService();
export default emailService;
